test/test.polar

```diff
@@ -15,6 +15,12 @@ builtinSpecializers(x: Float, "Float") if x > 0.0;
 builtinSpecializers(x: List, "List") if x = ["foo", *_rest];
 builtinSpecializers(x: Dictionary, "Dictionary") if x.foo = "foo";
 builtinSpecializers(x: String, "String") if x = "foo";
+# This should not match
```

Comment:
Maybe it should for JS, re: previous review comment about JS's permissiveness in returning `undefined` for nonexistent properties.

Reply:
It also checks that the value is 2.  Why would that match?

Reply:
![true](https://media.giphy.com/media/3orieXdGbfesO86UFy/source.gif)